lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -124,11 +112,26 @@ public function getExtension()
      */
     protected function getFilename($id)
     {
+        if ('\\' === DIRECTORY_SEPARATOR) {
```

Comment:
The problem with `bin2hex` everything is that it increases the filename by a factor of two. Most filesystem have a limit of 255 chars for filename/path components. So when using a cache key with a length of >=128, it will create an error. And we lose the advantage of having the cache key = filename for non-special chars.
In this case we could just as well use the hash for the filename as well.